timemachine/potentials/potential.py

```diff
@@ -13,7 +13,7 @@
 
 Precision = Any
 
-_P = TypeVar("_P", bound="Potential")
+_P = TypeVar("_P", bound="Potential", covariant=True)
```

Comment:
This allows mypy to treat e.g. `BoundPotential[HarmonicBond]` as an instance of `BoundPotential[Potential]`, allowing more types to be inferred without explicit annotation.